src/extension.ts

```diff
@@ -36,12 +36,12 @@ export let rMarkdownPreview: RMarkdownPreviewManager | undefined = undefined;
 
 // Called (once) when the extension is activated
 export async function activate(context: vscode.ExtensionContext): Promise<apiImplementation.RExtensionImplementation> {
-    if (!fs.existsSync(extDir)) {
-        fs.mkdirSync(extDir);
+    if (!fs.existsSync(extensionDirectory)) {
+        fs.mkdirSync(extensionDirectory);
     }
 
-    if (!fs.existsSync(tmpDir)) {
-        fs.mkdirSync(tmpDir);
+    if (!fs.existsSync(temporaryDirectory)) {
+        fs.mkdirSync(temporaryDirectory);
     }
```

Comment:
This creates these directories quite eagerly, i.e. as soon as the extension activates. Wouldn't it be better to create them only if they're actually used?

This could e.g. be implemented by adding a function that creates the directory (if necessary) and then returns its path

Reply:
Should now only be created when the variable is referred to:

```js
export const extensionDirectory = (): string => util.getDir(path.join(os.homedir(), '.vscode-R'));
export const temporaryDirectory = (): string => util.getDir(path.join(extensionDirectory(), 'tmp'));
```

Names tbd :)